smart_open/azure.py

```diff
@@ -15,8 +15,17 @@
 import smart_open.bytebuffer
 import smart_open.constants
 
-import azure.storage.blob
-import azure.core.exceptions
+import_error = """You are trying to use the Azure functionality of smart_open
```

Comment:
I don't like the idea of duplicating this across all of the modules.

The pattern is predictable:

- `azure.py` -> `pip install smart_open[azure]`
- `gcs.py` -> `pip install smart_open[gcs]`
- `s3.py` -> `pip install smart_open[aws]` (this one needs to change)
- ... and so on

So all you need to do is provide an informative message when handling the ImportError in transport.py.  Something like:

```python
try:
    ... # code that tries to import backend
except ImportError as error:
    logger.exception(error)
    logger.info('try `pip install smart_open[%s]` to install the missing dependencies' % submodule)
```

For `s3.py`, we should either rename the module to `aws.py` or rename the extras to `s3` to make things more predictable. I think the latter is easier than the former.

Reply:
Sure, I can do that. One thing that will change by moving this error handling to transport is that you won't get nice errors if you're using smart_open's service-specific modules from python code, e.g.

```python
from smart_open import gcs
```

As-is that code would show the suggested pip install error, but if moved it will revert to the normal GCS import error. Is that acceptable?

Reply:
I refactored the code to work as you suggest